network/retrieval/retrieve.go

```diff
@@ -346,6 +350,13 @@ func (r *Retrieval) handleRetrieveRequest(ctx context.Context, p *Peer, msg *Ret
 // we treat the chunk as a chunk received in syncing
 func (r *Retrieval) handleChunkDelivery(ctx context.Context, p *Peer, msg *ChunkDelivery) {
 	p.logger.Debug("retrieval.handleChunkDelivery", "ref", msg.Addr)
+	err := p.checkRequest(msg.Ruid, msg.Addr)
+	if err != nil {
+		unsolicitedChunkDelivery.Inc(1)
+		p.logger.Error("unsolicited chunk delivery from peer. dropping", "ruid", msg.Ruid, "addr", msg.Addr, "err", err)
```

Comment:
I think `dropping` can be deleted from this error log. `p.Drop` already logs that we are dropping the peer with the given reason.